### DISCUSSION TITLE
Tweak themes for "better" colors

### DISCUSSION THREAD

**Discussion by math-GH** [FreshRSS/FreshRSS#5205] [Show and tell]
>   I use alternative dark because I love dark reader themes and the normal dark is too low contrast for me. Alternative dark is a lot easier to navigate. The only downside is that there is too little color in this theme. Too much white-grey-black. I personally enjoy nicely color coded elements.

_Originally posted by @Zaczero in https://github.com/FreshRSS/FreshRSS/issues/4295#issuecomment-1465825657_

**Comment by math-GH** [comment]
@Zaczero

Thanks for your feedback.
Let me help you a bit to customize FreshRSS the way you like,

With a little bit work it is easy to improve the themes.

1. Install CustomCSS extension: https://github.com/FreshRSS/Extensions/tree/master/xExtension-CustomCSS
2. find out which colors you want to improve.

(here: Alternative Dark theme)
There is a bunch of color variables:
![grafik](https://user-images.githubusercontent.com/1645099/224788332-09b78c47-fcee-495b-9dae-1bcccdf592cd.png)

3. change the color in the CustomCSS extension.

For example to darken the side bar:
```
:root {
--background-color-light: #171717
}
```

Let me know if it helps you. Do not hesitate to ask questions :)

**Reply by daffydock** [reply]
Hi. I have done my own changes to the Dark theme and it was a bit of a pain to type, mostly as I was experimenting with a few things, which is fine. Yet it occurs to me that others may find it not as user friendly if they opt to tinker, like I did. 

Is there a possibility where you can paste the above list in text form for easier copy/pasting, as well?
Cheers.

**Reply by math-GH** [reply]
You can find the CSS file for the Dark theme here: https://github.com/FreshRSS/FreshRSS/blob/edge/p/themes/Dark/dark.css
and the frss.css here: https://github.com/FreshRSS/FreshRSS/blob/edge/p/themes/base-theme/frss.css

In general: the themes are store in `/p/themes/`

**Reply by daffydock** [reply]
@math-GH 

That's even better. Anyway to highlight this so others can find it easily as well?
Either way, thank you for letting me know.

**Reply by math-GH** [reply]
Indeed, an improved documentation would help.
This page needs an update with information about the CSS variables:
https://freshrss.github.io/FreshRSS/en/developers/04_Frontend/02_Design.html

Help is welcome

**Comment by andris155** [comment]
I don't like the color of notification-good-border-color (#0044cb) in the alternative dark theme. I think this color would be better for it: #0062b7

Current color
<img width="1003" alt="Screenshot 2023-03-13 224621" src="https://user-images.githubusercontent.com/8324198/224839394-2af7e5b5-4374-42fd-98ea-c19ddfc269f8.png">

Suggested color
<img width="1009" alt="Screenshot 2023-03-13 224545" src="https://user-images.githubusercontent.com/8324198/224839469-7ec27b45-a8c2-4038-a767-0ca8e4bd6771.png">

**Reply by math-GH** [reply]
Good idea. Made a PR to improve it: https://github.com/FreshRSS/FreshRSS/pull/5206

**Reply by math-GH** [reply]
it is now in the `edge` branch and will be delivered with the next release

**Reply by andris155** [reply]
Thanks!

**Comment by Zaczero** [comment]
In light themes you can choose between various primary colors. In dark themes it's not possible but I kinda understand the limitations. It would be great to see some sort of system for picking the primary color of your choice.

**Reply by Zaczero** [reply]
Maybe it would be possible to override CSS variables via dynamically generated style element? And this way a user would be able to choose any color they want from the settings.

**Reply by aledeg** [reply]
Maybe you could try with the help of customCSS extension.

**Reply by Zaczero** [reply]
Yes, but it would be cool to see a primary color picker in FreshRSS Themes V2 in some future :P, just giving some ideas.

**Comment by math-GH** [comment]
Just to mention: since 1.21.0 there is a "new" dark theme. It is a bit hidden: 
Use theme `Origine`/`Origine compact` + `enable `the auto dark mode + use a `dark browser` theme.

**Reply by Zaczero** [reply]
There should be a "Yes" option to use dark mode regardless.

**Reply by math-GH** [reply]
maybe it will be implemented later. See https://github.com/FreshRSS/FreshRSS/pull/4843#issuecomment-1343312523
